pkg/auth/device_code.go

```diff
@@ -54,15 +58,30 @@ func (deviceCode *DeviceCode) PollToken(auth0Token *Auth0Token) error {
 	data.Set("device_code", deviceCode.DeviceCode)
 	data.Set("grant_type", "urn:ietf:params:oauth:grant-type:device_code")
 
-	err := spinner.Poll(func() (bool, error) {
-		return fetchTokenFunc(auth0Token, data)
-	},
-		DEVICE_CODE_POLLING_INTERVAL,
-		DEVICE_CODE_POLLING_TIMEOUT)
+	fetchTokenFunc := func() (bool, error) {
+		err = auth0Token.Fetch(data)
+		if err == nil {
+			return true, nil
+		}
+
+		var auth0Err *Auth0Error
+		if errors.As(err, &auth0Err) {
+			if auth0Err.Type == "authorization_pending" {
+				return false, nil
+			}
+		}
+
+		return false, err
+	}
+
+	err = spinner.Poll(ctx, fetchTokenFunc, DEVICE_CODE_POLLING_INTERVAL, DEVICE_CODE_POLLING_TIMEOUT)
+
 	if err == nil {
 		return nil
 	}
 
+	spinner.StopFail()
```

Comment:
it might be called twice now - is that ok ? do we still need the stopFail call in the ui.ErrSpinnerTimeout case?

Reply:
fixed